### ISSUE TITLE
fzf relies on incorrect terminal wrapping behavour

### ISSUE THREAD

**Issue by sasq64** [junegunn/fzf#1033]
- Category
    - [x] fzf binary
    - [ ] fzf-tmux script
    - [ ] Key bindings
    - [ ] Completion
    - [ ] Vim
    - [ ] Neovim
    - [ ] Etc.
- OS
    - [ ] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Windows Subsystem for Linux
    - [ ] Etc.
- Shell
    - [ ] bash
    - [ ] zsh
    - [ ] fish

`fzf` seems to rely on that outputting a character at the last column will _not_ wrap to next line. This is true for modern terminal emulators that basically provides infinitely long lines to be able to do dynamic re wrapping of text but doesn't work on a basic xterm/vt100 emulation (which incidentally is what I am working on).

**Comment by junegunn** [comment]
Is there any way I can reproduce the problem on my macbook?

**Comment by sasq64** [comment]
Well... I don't currently know of any terminal emulators except mine that doesn't treat lines as infinitely long :)

Investigating this further it turns out that for instance zshell does this too in some cases with the line editor and custom prompts, so I guess this is just one of the many ways in which de facto behavior has kind of overridden the standard.

Luckily I came up with a work-around; Allow the cursor to be temporarily outside the right border, and don't wrap until the application actually tries to write text there -- it seems to work in all cases I tried.

I guess I'll close this, I was too hasty in thinking that this was an isolated thing that only fzf did.
